### PR TITLE
repoupdater: Extract Client interface

### PR DESCRIPTION
**Pull request by ryanslade**
This will make writing tests that need to assert we call repo updater
methods easier.

## Test plan

Existing tests

### CODE REVIEW

File: cmd/frontend/graphqlbackend/graphqlbackend.go [hunk 1]
```diff
@@ -550,7 +550,7 @@ type schemaResolver struct {
 	logger            log.Logger
 	db                database.DB
 	gitserverClient   gitserver.Client
-	repoupdaterClient *repoupdater.Client
+	repoupdaterClient *repoupdater.client
```
Comment (mrnugget) [inline cmd/frontend/graphqlbackend/graphqlbackend.go:553]: Pretty sure this won't work, since `client` is not exported?

Comment (ryanslade) [reply cmd/frontend/graphqlbackend/graphqlbackend.go:553]: Not ready for review yet, still a draft 😄

**Comment by ryanslade** [comment]
Closing this as I didn't realise that we already have mocks set up in that package, for example:
https://sourcegraph.sourcegraph.com/github.com/sourcegraph/sourcegraph@69fa190f1ec69a94eb10a9ed3d7e7099b191ad35/-/blob/internal/repoupdater/client.go?L132:4

I don't want to introduce a second way of doing things. For now I'll use what we already have.

File: cmd/frontend/graphqlbackend/graphqlbackend.go (hunk 1 above)
Comment (mrnugget) [reply cmd/frontend/graphqlbackend/graphqlbackend.go]: Is it? 😛 

screenshot_2022-12-20_16 22 06@2x

Comment (ryanslade) [reply cmd/frontend/graphqlbackend/graphqlbackend.go]: Oh, I assumed that it wouldn't trigger the review request until after you moved it out of draft mode.

**Pull request closed without merging**